style_editor.html

```diff
@@ -1178,6 +1178,26 @@
 var width;
 var height;
 
+function safeguardInputs() {
+  // Get all input elements
+  var inputs = document.querySelectorAll('input');
+
+  inputs.forEach(function(input) {
+    // Select the entire contents of the input box on click
```

Comment:
What exactly does this achieve?  For accessibility purpose, I'm just wary of changing default click behavior for an input.

Reply:
I just selects all on click instead of choosing an insert point and then needing to backspace or delete individual digits to achieve the new number. Just click and type.

Reply:
That was what it looked like.  I wonder if users on the whole will find this more or less helpful that default functionality.

Reply:
Because if there's a zero there, clicking to the left of it then deleting it would yield another zero in its place.
Clicking to the right of it would make the zero a leading number.
It just made most sense to click and have all selected ready for direct input.

Reply:
Maybe also reset the value to 0 on focusout

Reply:
I don't follow.

Reply:
Instead of having to force selecting the entire thing, and resetting to 0 the instant the user clears the input, you could do something like change
`function ArgChanged(ARG) {
  var N = ArgumentName_ENUM_BUILDER.value_to_name[ARG];
  var tag = FIND("ARGSTR_"+N);
  setARG(ARG, tag.value);
}`

to 

`function ArgChanged(ARG) {
  var N = ArgumentName_ENUM_BUILDER.value_to_name[ARG];
  var tag = FIND("ARGSTR_"+N);
  setARG(ARG, (tag.value || 0));
}`

Reply:
That doesn't work. deleting the value shows NaN.

Reply:
Huh.  Does the 0 have to be inside a string?  An empty value should be falsey, causing it to land on the 0.

Reply:
I don’t mean to press.  I’m just worried that the input trigger will create a bad user experience for anyone not expecting it.